broker: forward content.checkpoint-{get,put} RPCs to rank 0

chu11 opened the pull request:
Per a discussion in #4492, there was a mistake in PR #4463 were non-rank 0 brokers did not forward checkpoint RPCs to rank 0.  Instead, they were forwarded to rank 0's backing module.  It has little effect / meaning before the work in #4492, but felt it wise to split it out into its own PR as it solves a specific thing.  I also put some of the work from #4492 into this PR, since it made sense.  Most notably, the movement of the content-checkpoint code out of `content-cache` and into its own `content-checkpoint` files.  

Also added a helper lib for the content tests, b/c I was getting irritated constantly modifying all three content tests all of the time.

garlick commented on src/broker/content-cache.c:
```diff
@@ -645,7 +645,7 @@ void content_checkpoint_get_request (flux_t *h, flux_msg_handler_t *mh,
     const char *topic = "content-backing.checkpoint-get";
     const char *s = NULL;
-    const flux_msg_t *msgcpy = flux_msg_incref (msg);
+    const flux_msg_t *msgref = flux_msg_incref (msg);
```
I suggest instead of renaming the variable, we get rid of it and fix the refcounting problem on the error paths, as discussed in the other PR.

While you're at it, may as well fix the future leak that occurs when aux_set or then fails.

garlick commented on t/t0024-content-s3.t:
```diff
@@ -127,6 +127,20 @@ test_expect_success HAVE_JQ 'checkpoint-get foo returned rootref bar' '
         test_cmp rootref.exp rootref.out
 '
 
+test_expect_success HAVE_JQ 'checkpoint-put on rank 1 forwards to rank 0' '
+       o=$(checkpoint_put_msg rankone rankref) &&
+       jq -j -c -n ${o} | flux exec -r 1 ${RPC} content.checkpoint-put
+'
+
+test_expect_success HAVE_JQ 'checkpoint-get on rank 1 forwards to rank 0' '
+       echo rankref >rankref.exp &&
+       o=$(checkpoint_get_msg rankone) &&
+       jq -j -c -n ${o} \
+	   | flux exec -r 1 ${RPC} content.checkpoint-get \
+	   | jq -r .value | jq -r .rootref > rankref.out &&
+       test_cmp rankref.exp rankref.out
+'
+
```
These tests really don't need to be repeated in every backing store since the code they are testing is in the broker not the backing stores...  Just pick one?

garlick reviewed (approved):
Couple of nits, otherwise good.

chu11 commented:
re-pushed, removed those extra tests per comments above.  Instead of the `msgcpy` -> `msgref` commit, I just added a commit that refactored things to remove the temp variable and fixup the flux future double decref issue.

garlick reviewed (approved):
LGTM!
